shared/src/androidMain/kotlin/dev/johnoreilly/confetti/work/SessionNotificationWorker.kt

```diff
@@ -18,15 +21,23 @@ class SessionNotificationWorker(
 ) : CoroutineWorker(context, workerParams) {
 
     override suspend fun doWork(): Result {
-        notifier.sendNotification()
-        return Result.success()
+        if (runAttemptCount > 3) {
+            return Result.failure()
+        }
+
+        return try {
+            notifier.sendNotification()
+            Result.success()
+        } catch (e: Throwable) {
+            Result.retry()
```

Comment:
That should make the worker more robust by not allowing the task to retry up to 3 times if a failure occurs.